MeretzLib/Apple/iOS/Meretz/Meretz/Meretz.h

```diff
@@ -86,19 +86,46 @@ typedef NS_ENUM(NSInteger, MeretzTaskStatus)
 	@property (nonatomic, retain) NSArray* Items;
 @end
 
+/* $FUTURE
 @interface MeretzVendorUserProfileResult : MeretzResult
 	// # of points user can spend currently, will be an integer
 	@property (nonatomic, retain) NSNumber* UsablePoints;
 	// total # of points the user has, will be an integer >= usablePoints
 	@property (nonatomic, retain) NSNumber* TotalPoints;
 @end
+*/
+
+/* ---------- delegate protocol */
+
+@protocol MeretzDelegate<NSObject>
+@required
+	// called when a user connection with the Meretz back-end completes
+	// if successful (result.Success==TRUE), result.AccessToken should be saved and used for all future API calls
+	- (void) didVendorUserConnectFinish:(MeretzVendorUserConnectResult *)result;
+	// called when a user disconnection from the Meretz back-end completes
+	// if successful (result.Success==TRUE), the user is no longer connected with the Meretz backend and
+	// any saved access token is now invalid.
+	- (void) didVendorUserDisconnectFinish:(MeretzResult *)result;
+	// called when a vendorConsumeWithinRange task finishes
+	// if successful (result.Success==TRUE), result.Items contains all owned items within specified date range.
+	- (void) didVendorConsumeWithinRangeFinish:(MeretzVendorConsumeResult *)result;
+	// called when a vendorConsumeGetNew task finishes
+	// if successful (result.Success==TRUE), result.Items contains any newly acquired (un-consumed) items.
+	-(void) didVendorConsumeGetNewFinish:(MeretzVendorConsumeResult *)result;
+	// called when a vendorConsumeAcknowledge task finishes
+	// if successful (result.Success==TRUE), result.Items contains the acknowledged (consumed) items.
+	-(void) didVendorConsumeAcknowledgeFinish:(MeretzVendorConsumeResult *)result;
+@end
 
 // Meretz API interface
 
 @interface Meretz : NSObject
 
 	// call this to initialize Meretz for your organization's application
```

Comment:
SDK can now be initialized using a saved access key and delegate object (both are optional)